public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -59,7 +59,18 @@ import { increasingInterval } from './utils/rxjs/increasingInterval';
 import { toTestingStatus } from '@grafana/runtime/src/utils/queryResponse';
 import { addDataLinksToLogsResponse } from './utils/datalinks';
 import { runWithRetry } from './utils/logsRetry';
-import { CompletionItemProvider } from './cloudwatch-sql/completion/CompletionItemProvider';
+
+import { SQLCompletionItemProvider } from './cloudwatch-sql/completion/CompletionItemProvider';
+import { getSuggestionKinds as getSQLSuggestionKinds } from './cloudwatch-sql/completion/suggestionKind';
+import { getStatementPosition as getSQLStatementPosition } from './cloudwatch-sql/completion/statementPosition';
+import cloudWatchSqlLanguageDefinition from './cloudwatch-sql/definition';
+import { SQLTokenType } from './cloudwatch-sql/completion/types';
+
+import { MetricMathCompletionItemProvider } from './metric-math/completion/CompletionItemProvider';
+import { getSuggestionKinds as getMetricMathSuggestionKinds } from './metric-math/completion/suggestionKind';
+import { getStatementPosition as getMetricMathStatementPosition } from './metric-math/completion/statementPosition';
+import cloudWatchMetricMathLanguageDefinition from './metric-math/definition';
+import { MetricMathTokenType } from './metric-math/completion/types';
```

Comment:
Instead of importing all these things (line 63-73), can't they be imported in their respective CompletionItemProvider and then pass it along to the super constructor? Then we only need to instantiate MetricMathCompletionItemProvider and SQLCompletionItemProvider with a reference to _this_  and template here.

Reply:
@sunker I realized there's really no reason to pass in these args at all since they are static, so I reworked it a bit, hopefully it makes sense, let me know if you want me to walk through any of the changes!